core/backend/src/IModelDb.ts

```diff
@@ -2931,16 +2917,23 @@ export class SnapshotDb extends IModelDb {
     // unref timer, so it doesn't prevent a process from shutting down.
     snapshot._restartDefaultTxnTimer = setTimeout(() => {
       snapshot.restartDefaultTxn();
-    }, (10 * 60) * 1000).unref(); // 10 * 60 is 10 minutes in seconds, then converted to milliseconds (* 1000);
-    snapshot._refreshSas = new RefreshV2CheckpointSas(container.accessToken, checkpoint.reattachSafetySeconds);
+    }, (10 * 60) * 1000).unref(); // 10 minutes
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    snapshot._refreshSas = new RefreshV2CheckpointSas(snapshot.nativeDb.cloudContainer!.accessToken, checkpoint.reattachSafetySeconds);
     return snapshot;
   }
 
+  // Open a Checkpoint directly from a cloud container.
+  public static async openCheckpoint(checkpoint: CheckpointProps): Promise<SnapshotDb> {
```

Comment:
Should this be marked @beta? 

Should we also consider making prefetching more visible? I imagine an IPC app may want to be able to prefetch / know how to trigger one.

Reply:
Yes, i do think we should add documentation about how prefetching works (and of how you indicate that a checkpoint should be prefetched in particular). I leave it to you whether you want to add that to this PR or another one.

Along those lines, I'm not sure I understand the logic of skipping prefetches for big iModels. It seems to me that's where it will help the most (although I recognize the cachefile will fill/spill quicker). How did we arrive at the default 2G size? That seems way too small to me, but perhaps a better approach is to limit the time a prefetch can take, or maybe some other way to determine when to stop prefetching. For desktops, the default should be different. This topic should be researched/tested and I don't volunteer for that.

Reply:
I will address that(documentation + prefetching desktop) in another PR then. I agree that in the desktop case we definitely need to reconsider things and should prefetch probably everything (as long as user requests it and they understand the ramifications, which all comes with the documentation part)

2G default was based on poorer performance across a node (not just the imodel thats being prefetched) when prefetching XL iModels. I think we can probably bump up that 2G in our cluster. Additionally I think just prefetching one XL iMdoel slowed down the time it took for its requests to finish in the stresstest we were running, but I'd have to look back at my notes.

I also agree we could probably use a better approach in the daemon case. I still like the idea of having the daemon maximize its throughput and prefetching everything but backing off (especially on the larger iModels) when it needs to in the case of higher activity. Feel free to resolve this comment unless you want to continue the discussion.